### PR TITLE
Fix url regex #15

### PR DESCRIPTION
**Pull request by iwi**
fix issue #15

### CODE REVIEW

File: tests/test_message.py [hunk 1]
```diff
@@ -4,17 +4,49 @@
...
 # test url detection
 
-def test_url_detection_of_https():
+def test_basic_url_detection_of_https():
     expected = "https://example.org"
+    assert message.extract_url("foo <https://example.org>") == expected
 
-    assert message.extract_url("foo https://example.org") == expected
+def test_hash_in_url():
+    expected = "https://#.org"
```
Comment (arnau) [inline tests/test_message.py]: I understand that there is no need to enforce strictly correct URLs at the regex level but at least the tests should show valid URLs.

So, I'd rather use

```
https://example.org#foo
```

Comment (iwi) [inline tests/test_message.py]: ok

File: tests/test_message.py [hunk 2]
```diff
@@ -6,47 +6,57 @@
...
+    expected = "http://foo.org/foo+bar"
+    text = "<http://foo.org/foo+bar>"
+    assert message.extract_url(text) == expected
 
 
 def test_equal_in_url():
-    expected = "https://=.org"
-    assert message.extract_url("<https://=.org>") == expected
+    expected = "https://wikipedia.org/title=foo"
+    text = "<https://wikipedia.org/title=foo>"
```
Comment (arnau) [inline tests/test_message.py]: A common URL with an equal is `http://example.org?q=foo`

File: tests/test_message.py [hunk 3]
```diff
@@ -6,47 +6,57 @@
...
+    assert message.extract_url(text) == expected
 
 
 def test_qmark_in_url():
-    expected = "https://?.org"
-    assert message.extract_url("<https://?.org>") == expected
+    expected = "http://wik.org/index.php?title"
+    text = "<http://wik.org/index.php?title>"
```
Comment (arnau) [inline tests/test_message.py]: To test a question mark I'd stick to `http://example.org?`

File: tests/test_message.py [hunk 4]
```diff
@@ -6,47 +6,57 @@
...
+    assert message.extract_url(text) == expected
 
 
 def test_and_in_url():
-    expected = "https://&.org"
-    assert message.extract_url("<https://&.org>") == expected
+    expected = "https://foo.org/alpha&beta"
+    text = "<https://foo.org/alpha&beta>"
```
Comment (arnau) [inline tests/test_message.py]: The normal form is `http://example.org?foo=1&bar=2`

File: tests/test_message.py (hunk 2 above)
Comment (iwi) [inline tests/test_message.py]: I saw that but wondered if it made sense to have a question mark on the same one

File: tests/test_message.py (hunk 4 above)
Comment (iwi) [inline tests/test_message.py]: ditto

**Pull request merged**